File: exercise-03-build-your-own-tooling/mcp-exercises/nodejs-starter/src/features/users/user-routes.ts
// Starter routes file — used as context during Exercise 3 (Structure)
//
// Exercise 3:
//   Ask the AI to create a new "orders" feature.
//   Watch where it puts the files — with and without the MCP server.

import express from "express";

const router = express.Router();

router.get("/:id", async (req, res) => {
  res.json({ id: req.params.id, name: "Alice Johnson", email: "alice@example.com" });
});

router.get("/", async (_req, res) => {
  res.json([
    { id: "1", name: "Alice Johnson" },
    { id: "2", name: "Bob Smith" },
  ]);
});

export default router;

File: exercise-03-build-your-own-tooling/mcp-exercises/python-starter/app/users/user_routes.py
# Starter routes file — used as context during Exercise 3 (Structure)
#
# Exercise 3:
#   Ask the AI to create a new "orders" feature.
#   Watch where it puts the files — with and without the MCP server.

from fastapi import APIRouter

router = APIRouter(prefix="/users", tags=["users"])


@router.get("/{user_id}")
async def get_user(user_id: str) -> dict[str, str]:
    return {"id": user_id, "name": "Alice Johnson", "email": "alice@example.com"}


@router.get("/")
async def list_users() -> list[dict[str, str]]:
    return [
        {"id": "1", "name": "Alice Johnson"},
        {"id": "2", "name": "Bob Smith"},
    ]
